internal/v1/handler.go

```diff
@@ -86,17 +88,25 @@ func (h *Handlers) GetDistributions(ctx echo.Context) error {
 	dr := h.server.distroRegistry(ctx)
 	userID, err := h.server.getIdentity(ctx)
 	if err != nil {
-		return err
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to get user identity: %v", err))
 	}
 
+	distributions, err := h.filterDistributions(userID.OrgID(), *dr)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to filter distributions").SetInternal(err)
+	}
+	return ctx.JSON(http.StatusOK, distributions)
+}
+
+func (h *Handlers) filterDistributions(orgID string, distroMap distribution.DistroRegistry) (DistributionsResponse, error) {
```

Comment:
This could also be a function, not a method if we pass the allowList as a parameter.
It would be sligtly cleaner in my opinion (we would not pollute the handlers object with helper methods), but again this is just a question of taste.
```suggestion
func filterDistributions(orgID string, distroMap distribution.DistroRegistry, allowList *<TheAllowListType>) (DistributionsResponse, error) {
```

Reply:
this one look more complicated I prefer to keep it as is